fpdf/table.py

```diff
@@ -0,0 +1,337 @@
+        if lines_heights_only:
+            return lines_heights
+
+    def _get_col_width(self, i, j, colspan=1):
+        if not self._col_widths:
+            cols_count = self._rows[i].cols_count
+            return colspan * (self._width / cols_count)
+        if isinstance(self._col_widths, Number):
+            return colspan * self._col_widths
+        if j >= len(self._col_widths):
+            raise ValueError(
+                f"Invalid .col_widths specified: missing width for table() column {j + 1} on row {i + 1}"
+            )
+        # pylint: disable=unsubscriptable-object
+        col_width = 0
+        for k in range(j, j + colspan):
+            col_ratio = self._col_widths[k] / sum(self._col_widths)
+            col_width += col_ratio * self._width
+        return col_width
+
+    def _get_lines_heights_per_cell(self, i) -> List[List[int]]:
+        row = self._rows[i]
+        lines_heights = []
+        for j in range(len(row.cells)):
+            lines_heights.append(
+                self._render_table_cell(
+                    i,
+                    j,
+                    cell_line_height=self._line_height,
+                    row_height=self._line_height,
+                    lines_heights_only=True,
+                )
+            )
+        return lines_heights
+
+
+class Row:
+    "Object that `Table.row()` yields, used to build a row in a table"
+
+    def __init__(self):
+        self.cells = []
+
+    @property
+    def cols_count(self):
+        return sum(cell.colspan for cell in self.cells)
+
+    def cell(
+        self, text="", align=None, style=None, img=None, img_fill_width=False, colspan=1
+    ):
+        """
+        Adds a cell to the row.
+
+        Args:
+            text (str): string content, can contain several lines.
+                In that case, the row height will grow proportionally.
+            align (str, fpdf.enums.Align): optional text alignment.
+            style (fpdf.fonts.FontStyle): optional text style.
+            img: optional. Either a string representing a file path to an image,
+                an URL to an image, an io.BytesIO, or a instance of `PIL.Image.Image`.
+            img_fill_width (bool): optional, defaults to False. Indicates to render the image
+                using the full width of the current table column.
+            colspan (int): optional number of columns this cell should span.
+        """
+        if text and img:
+            raise NotImplementedError(
+                "fpdf2 currently does not support inserting text with an image in the same table cell."
+                "Pull Requests are welcome to implement this 😊"
+            )
+        cell = Cell(text, align, style, img, img_fill_width, colspan)
+        self.cells.append(cell)
+        return cell
+
+
+@dataclass
+class Cell:
+    "Internal representation of a table cell"
+    text: str
+    align: Union[str, Align]
+    style: FontStyle
+    img: str
+    img_fill_width: bool
```

Comment:
Do we want an alternative `image_fill_height` as well? (possible future enhancement)

Reply:
The current default behaviour is ti fill the row height,
with the row height being defined by the other cells in the current row.

A future enhancement would be to give the user explicit control over a row height,
but I think it would be better to wait for your TextRegions before defining how best to implement this.